OpenKeychain/src/main/java/org/sufficientlysecure/keychain/receiver/NetworkReceiver.java

```diff
@@ -20,29 +20,30 @@ public void onReceive(Context context, Intent intent) {
         ConnectivityManager conn = (ConnectivityManager)
                 context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo networkInfo = conn.getActiveNetworkInfo();
+        boolean isTypeWifi = (networkInfo.getType() == ConnectivityManager.TYPE_WIFI);
+        boolean isConnected = networkInfo.isConnected();
 
-        if (networkInfo != null && networkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
+        if (isTypeWifi && isConnected) {
 
             // broadcaster receiver disabled
             setWifiReceiverComponent(false, context);
             Intent serviceIntent = new Intent(context, KeyserverSyncAdapterService.class);
-            serviceIntent.setAction(KeyserverSyncAdapterService.ACTION_UPDATE_ALL);
+            serviceIntent.setAction(KeyserverSyncAdapterService.ACTION_SYNC_NOW);
```

Comment:
`ACTION_SYNC_NOW` seems to resume `KeyserverSyncAdapterService` more reliably when wifi is restored.